### PR TITLE
lint: add param-name checker

### PR DESCRIPTION
**Pull request by Quasilyte**
Also modify kfulint/main.go and lint.go to make
it possible to implement first checker.

### CODE REVIEW

File: lint/param-name_checker.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+	warnings []Warning
+}
+
+// NewParamNameChecker returns initialized checker for Go functions param names.
+func NewParamNameChecker(ctx *Context) *ParamNameChecker {
+	return &ParamNameChecker{
+		ctx: ctx,
+		loudNames: map[string]bool{
+			"IN":    true,
```
Comment (fexolm) [inline lint/param-name_checker.go]: Is there any reason to use map instead on set here?

File: lint/param-name_checker.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+//    treat it as "redundantly exported".
+//    Suggests to use non-capitalized identifier.
+func (c *ParamNameChecker) Check(f *ast.File) []Warning {
+	c.warnings = c.warnings[:0]
+	for _, decl := range collectFuncDecls(f) {
+		for _, param := range collectFuncParams(decl) {
+			for _, id := range param.Names {
+				switch {
+				case c.loudNames[id.Name]:
+					c.warnLoud(id)
```
Comment (fexolm) [inline lint/param-name_checker.go]: Maybe we need also add camel case check here?

File: lint/util.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	for _, decl := range f.Decls {
+		if decl, ok := decl.(*ast.FuncDecl); ok {
+			decls = append(decls, decl)
+		}
+	}
+	return decls
+}
+
+func collectFuncParams(decl *ast.FuncDecl) []*ast.Field {
+	var params []*ast.Field
```
Comment (fexolm) [inline lint/util.go]: It seems line that this function will work only in param_name_checker. Do we really need array with recv, params and results in global case? Maybe we should move this function in param_name_checker or slpit it into 3 functions?

File: lint/param-name_checker.go (hunk 2 above)
Comment (Quasilyte) [reply lint/param-name_checker.go]: `golint` does that.

File: lint/param-name_checker.go (hunk 1 above)
Comment (Quasilyte) [reply lint/param-name_checker.go]: There are 2 idioms for "sets" in Go:
- `map[T]bool`
- `map[T]struct{}`
The latter was more efficient in the past, not sure it's accurate anymore.
The fist one is preferred, you can check if element is in set by simply doing `m[key]`.

No "real" sets are builtin or inside stdlib.

File: lint/util.go (hunk 3 above)
Comment (Quasilyte) [reply lint/util.go]: Good point.
Fixed.

**Comment by Quasilyte** [comment]
We could force checkers to return `([]Warning, error)` and avoid panic for "exceptional situations", but:
1. I think most checkers will always return `nil` error.
2. Even if they want to return some kind of fatal error, it could be inconvenient to do so due to complications with `ast.Walk` and functions alike (can't just return from containing function while traversing AST with callback function).

**Pull request merged**